nifi-nar-bundles/nifi-kudu-bundle/nifi-kudu-processors/src/main/java/org/apache/nifi/processors/kudu/ScanKudu.java

```diff
@@ -0,0 +1,406 @@
+@Tags({"kudu", "scan", "fetch", "get"})
+@CapabilityDescription("Scans rows from a Kudu table with an optional list of predicates")
+@WritesAttributes({
+        @WritesAttribute(attribute = "kudu.table", description = "The name of the Kudu table that the row was fetched from"),
+        @WritesAttribute(attribute = "mime.type", description = "Set to application/json when using a Destination of flowfile-content, not set or modified otherwise"),
+        @WritesAttribute(attribute = "kudu.rows.count", description = "Number of rows in the content of given flow file"),
+        @WritesAttribute(attribute = "scankudu.results.found", description = "Indicates whether at least one row has been found in given Kudu table with provided predicates. "
+                + "Could be null (not present) if transfered to FAILURE")})
+public class ScanKudu extends AbstractKuduProcessor {
+
+    static final Pattern PREDICATES_PATTERN = Pattern.compile("\\w+((<=|>=|[=<>])(\\w|-)+)?(?:,\\w+((<=|>=|[=<>])(\\w|-)+)?)*");
+    static final Pattern COLUMNS_PATTERN = Pattern.compile("\\w+((\\w)+)?(?:,\\w+((\\w)+)?)*");
+
+    protected static final PropertyDescriptor TABLE_NAME = new PropertyDescriptor.Builder()
+            .name("table-name")
+            .displayName("Table Name")
+            .description("The name of the Kudu Table to put data into")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .build();
+
+  static final PropertyDescriptor PREDICATES =
+      new PropertyDescriptor.Builder()
+          .name("Predicates")
+          .description("A comma-separated list of Predicates,"
+              + "EQUALS: \"(colName)=(value)\","
+              + "GREATER: \"(colName)<(value)\","
+              + "LESS: \"(colName)>(value)\","
+              + "GREATER_EQUAL: \"(colName)>=(value)\","
+              + "LESS_EQUAL: \"(colName)<=(value)\"")
+          .required(false)
+          .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+          .addValidator(StandardValidators.createRegexMatchingValidator(PREDICATES_PATTERN))
+          .build();
+
+    static final PropertyDescriptor PROJECTED_COLUMNS = new PropertyDescriptor.Builder()
+            .name("Projected Column Names")
+            .description("A comma-separated list of \"<column>\" names to return when scanning, default all.")
+            .required(false)
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .addValidator(StandardValidators.createRegexMatchingValidator(COLUMNS_PATTERN))
+            .build();
+
+    protected static final PropertyDescriptor BATCH_SIZE = new PropertyDescriptor.Builder()
+            .name("Batch Size")
+            .description("The maximum number of RowResults to process in a single execution, between 1 - 100000. " +
```

Comment:
I think this is still misleading a bit as it's not the number or results in a single execution but the number of results in an output flowfile. Multiple flowfiles can be created in a single execution - which is in fact a good thing to indicate at least here.